src/js/components/Form/Form.js

```diff
@@ -124,29 +124,75 @@ const Form = forwardRef(
       [onChange],
     );
 
-    const useFormContext = (name, componentValue, defaultComponentValue) => {
-      const valueData = (name && value[name]) || defaultComponentValue;
-      const [data, setData] = useState(
-        componentValue !== undefined ? componentValue : valueData,
-      );
-      if (componentValue !== undefined) {
-        if (componentValue !== data) {
-          setData(componentValue);
-          if (name) update(name, componentValue);
-        } else if (name && value[name] === undefined) {
-          update(name, componentValue, true);
+    // There are three basic models of handling form input value state:
+    //
```

Comment:
Excellent comments!  A summary of these 3 models needs to find its way into the `Form` doumentation too.

With this PR, for the _Form Controlled _ model, does the `Form onChange` handler now only get called for _user-input changes_, as opposed to _controlled changes_, to the input fields (see issue #3935)?  If so, then the `Form onChange` documentation will need to also reflect that.

Reply:
>Excellent comments! A summary of these 3 models needs to find its way into the Form doumentation too.

It's getting there https://v2.grommet.io/form